### DISCUSSION TITLE
[Improvement]: Allow us to add custom Dockerfiles to docker compose and service deployments

### DISCUSSION THREAD

**Discussion by peaklabs-dev** [coollabsio/coolify#3156] [Improvement Requests]
### Description

At the moment when we want to modife the default docker image bein used in service deplyment or when using a docker compose file we have to build our own image and upload it to docker hub or upload a dockerifle onto the right folder in the server. This all is too complex

### Minimal Reproduction (if possible, example repository)

Ther should be a way that we can add a docker compose file with the folwoing information for exmaple:
```yml
wordpress:
    build:
      context: .
.............
```
And then in the deployment tab for all services and all docker compose deployments there is a field where we can add a custom dockerfile -> and also a checkbox we have to check if we want to use the costume dockerfile.
Steps to implement:
- [ ] Allow us to use a custom dockerfile for services and docker compose deployments -> So the full build settings would need to be made available in the services and docker compose deployment types
- [ ] (Add a checkbox to let us use our own docker file.)
- [ ] Below the checkbox, have a field that lets us add our own dockerfile (could look like this) -> This will be added to services and docker compose based deployments -> The full build settings are needed for that:
![image](https://github.com/coollabsio/coolify/assets/122374094/91cc18a2-2f30-4f1e-99f7-1c545928aefc)

### Version

-

**Comment by enesbala5** [comment]
I am using a command like this to build the images:

docker compose --env-file .env --env-file .env.prod -f docker-compose.yaml -f docker-compose.prod.yaml build

I am not sure if it's useful for you until the feature is implemented to the interface.

You can also create a script and call it directly. Like the one below:

cd bin && ./deploy.sh prod build

and 

cd bin && ./deploy.sh prod up -d

**Comment by peaklabs-dev** [comment]
Same thing the other way around would also be nice:

- [ ] When deploying with a dockerfile in a git repo, let us also select the yml file in the repo --> So we can have yml and dockerfile in one repo folder and use both to build an image and then deploy services with the docker compose file and using that image.
